sdk/metric/benchmark_test.go

```diff
@@ -0,0 +1,152 @@
+	}
+}
+
+func BenchmarkCounterAddOneAttr(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		cntr.Add(ctx, 1, attribute.String("K", "V"))
+	}
+}
+
+func BenchmarkCounterAddOneInvalidAttr(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		cntr.Add(ctx, 1, attribute.String("", "V"), attribute.String("K", "V"))
+	}
+}
+
+func BenchmarkCounterAddManyAttrs(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		cntr.Add(ctx, 1, attribute.Int("K", i))
+	}
+}
+
+func BenchmarkCounterAddManyInvalidAttrs(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		cntr.Add(ctx, 1, attribute.Int("", i), attribute.Int("K", i))
+	}
+}
+
+func BenchmarkCounterAddManyFilteredAttrs(b *testing.B) {
+	vw, _ := view.New(view.WithFilterAttributes(attribute.Key("K")))
+
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr, vw))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		cntr.Add(ctx, 1, attribute.Int("L", i), attribute.Int("K", i))
+	}
+}
+
+func BenchmarkCounterCollectOneAttr(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		cntr.Add(ctx, 1, attribute.Int("K", 1))
+
+		_, _ = rdr.Collect(ctx)
+	}
+}
+
+func BenchmarkCounterCollectTenAttrs(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < 10; j++ {
+			cntr.Add(ctx, 1, attribute.Int("K", j))
+		}
+		_, _ = rdr.Collect(ctx)
+	}
+}
+
+func BenchmarkCounterCollectTenAttrsTenTimes(b *testing.B) {
+	ctx := context.Background()
+	rdr := NewManualReader()
+	provider := NewMeterProvider(WithReader(rdr))
+	b.ReportAllocs()
+
+	cntr, _ := provider.Meter("test").SyncInt64().Counter("hello")
+
+	for i := 0; i < b.N; i++ {
+		for k := 0; k < 10; k++ {
+			for j := 0; j < 10; j++ {
+				cntr.Add(ctx, 1, attribute.Int("K", j))
+			}
+			_, _ = rdr.Collect(ctx)
+		}
+	}
+}
```

Comment:
This is unnecessary, it only serves to divide `b.N` by ten. It is equivalent to `BenchmarkCounterCollectTenAttrs`.

Reply:
Yes, sorry -- copy and pasted mistake.